apptools/preferences/tests/test_preferences_helper.py

```diff
@@ -245,16 +253,22 @@ class AcmeUIPreferencesHelper(PreferencesHelper):
         self.assertEqual(second_unicode_str, helper.description)
         self.assertEqual(second_unicode_str, p.get('acme.ui.description'))
 
-        p.save(self.example)
-
-        p.load(self.example)
-        self.assertEqual(second_unicode_str, p.get('acme.ui.description'))
-        self.assertEqual(u'True', p.get('acme.ui.visible'))
-        self.assertTrue(helper.visible)
-
-        # reset the original description and save the example file
-        helper.description = original_description
-        p.save(self.example)
+        # Save it to another file.
+        tmp = os.path.join(self.tmpdir, 'tmp.ini')
+        p.save(tmp)
+
+        try:
+            # Load it into a new node.
+            p = Preferences()
+            p.load(tmp)
+            self.assertEqual(second_unicode_str, p.get('acme.ui.description'))
+            self.assertEqual(u'True', p.get('acme.ui.visible'))
+            self.assertTrue(helper.visible)
+        finally:
+            # Clean up!
+            os.remove(tmp)
```

Comment:
This is why we more often see `shutil.rmtree` in the tear down, so that you don't have to make the test code more complicated by making sure each file is removed. e.g. We won't need this `os.remove` here if teardown is calling `shutil.rmtree`. All files in the temporary directory will be deleted no matter what. If you don't mind making this one more change to use `rmtree`, it would make writing new tests easier in the future.

Reply:
yep, sure thing